app/cli/plugins/interface.go

```diff
@@ -0,0 +1,102 @@
+//
+// Copyright 2025 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugins
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/spf13/viper"
+)
+
+// Plugin is the interface that plugins must implement.
+type Plugin interface {
+	// Exec executes a command within the plugin
+	Exec(ctx context.Context, command string, arguments map[string]any) (ExecResult, error)
```

Comment:
can you pass flags?

Reply:
We now have:
```
Exec(ctx context.Context, config plugins.PluginConfig)
```

With `PluginConfig` looking like this:

```
type PluginConfig struct {
	Command         string
	Args            []string
	Flags           map[string]any
	PersistentFlags map[string]SimpleFlag
}
```